tagless-final/src/main/java/com/github/jorgecastillo/kotlinandroid/tagless/runtime/ui/SuperHeroDetailActivity.kt

```diff
@@ -6,22 +6,18 @@ import android.os.Bundle
 import android.support.design.widget.Snackbar
 import android.support.v7.app.AppCompatActivity
 import android.widget.Toast
-import arrow.effects.ev
+import arrow.effects.extract
 import com.github.jorgecastillo.kotlinandroid.R
 import com.github.jorgecastillo.kotlinandroid.R.string
 import com.github.jorgecastillo.kotlinandroid.tagless.algebras.ui.SuperHeroDetailView
 import com.github.jorgecastillo.kotlinandroid.tagless.algebras.ui.extensions.loadImageAsync
 import com.github.jorgecastillo.kotlinandroid.tagless.algebras.ui.model.SuperHeroViewModel
-import com.github.jorgecastillo.kotlinandroid.tagless.algebras.ui.presentation
-import com.github.jorgecastillo.kotlinandroid.tagless.runtime.instances.TargetRuntime
-import kotlinx.android.synthetic.main.activity_detail.appBar
-import kotlinx.android.synthetic.main.activity_detail.collapsingToolbar
-import kotlinx.android.synthetic.main.activity_detail.description
-import kotlinx.android.synthetic.main.activity_detail.headerImage
+import com.github.jorgecastillo.kotlinandroid.tagless.runtime.instances.instances
+import kotlinx.android.synthetic.main.activity_detail.*
 
 class SuperHeroDetailActivity : AppCompatActivity(), SuperHeroDetailView {
 
-    val pres = presentation<TargetRuntime>()
+    val pres = instances().presentation()
```

Comment:
Could you call this injectInstances() for didactic purposes? People is used to call `component.inject(this)` when using dagger. the `DaggerInstances.create()` is injecting them?

Reply:
`component.inject(this)` I don't do that. Especially with Kotlin ;p

Reply:
It's only required if you want to do property injection.

Reply:
Dagger already provides it's own implementation of lazy injection so you don't need to make it lazy on code where it's used. Also, isn't `instances()` providing the component itself by calling `DaggerInstances.create()`? Or is our `DaggerInstances` class performing injection there also?

Reply:
`DaggerInstances` is the materialized graph. `@Lazy` in Dagger has different semantics in the sense that it is used to provide lazy instances but in this case we are talking about the entire DI graph being lazily initialized since it doesn't really matter much here because all this app does uses the instances in the graph.